enterprise/server/githubauth/githubauth.go

```diff
@@ -0,0 +1,326 @@
+package githubauth
+
+import (
+	"context"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/githubapp"
+	"github.com/buildbuddy-io/buildbuddy/server/backends/github"
+	"github.com/buildbuddy-io/buildbuddy/server/environment"
+	"github.com/buildbuddy-io/buildbuddy/server/interfaces"
+	"github.com/buildbuddy-io/buildbuddy/server/tables"
+	"github.com/buildbuddy-io/buildbuddy/server/util/authutil"
+	"github.com/buildbuddy-io/buildbuddy/server/util/claims"
+	"github.com/buildbuddy-io/buildbuddy/server/util/cookie"
+	"github.com/buildbuddy-io/buildbuddy/server/util/flagutil"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+	"github.com/golang-jwt/jwt"
+	"github.com/google/uuid"
+)
+
+const (
+	githubIssuer   = "https://github.com"
+	contextUserKey = "auth.githubuser"
+	loginPath      = "/login/github/"
+	authPath       = "/auth/github/"
+	jwtDuration    = 24 * 365 * time.Hour
+)
+
+var (
+	jwtKey = flagutil.New("github.jwt_key", "", "The key to use when signing JWT tokens for github auth.", flagutil.SecretTag)
+)
+
+type githubAuthenticator struct {
+	env environment.Env
+}
+
+func NewGithubAuthenticator(env environment.Env) *githubAuthenticator {
+	return &githubAuthenticator{
+		env: env,
+	}
+}
+
+func IsEnabled(env environment.Env) bool {
+	return githubapp.IsEnabled() && *jwtKey != ""
+}
+
+func (a *githubAuthenticator) Login(w http.ResponseWriter, r *http.Request) error {
+	if r.URL.Path != loginPath {
+		return status.UnauthenticatedError("not a github login")
+	}
+	a.handler().StartAuthFlow(w, r, "/auth/github/")
```

Comment:
should this just reference the authPath const?

Reply:
Done